module_utils/network_lsr/argument_validator.py

```diff
@@ -2292,14 +2296,34 @@ def _validate_post(self, value, name, result):
                             ),
                         )
                 if connection["parent"]:
-                    if not ArgUtil.connection_find_by_name(
+                    parent_connection = ArgUtil.connection_find_by_name(
                         connection["parent"], result, idx
-                    ):
+                    )
+                    if not parent_connection:
                         raise ValidationError(
                             name + "[" + str(idx) + "].parent",
                             "references non-existing 'parent' connection '%s'"
                             % (connection["parent"]),
                         )
+
+                    if (
+                        connection["type"] == "infiniband"
+                        and connection["infiniband"]["p_key"] != -1
+                        and connection["interface_name"]
+                    ):
+                        virtual_interface_name = "{0}.{1:0>4x}".format(
+                            parent_connection["interface_name"],
+                            connection["infiniband"]["p_key"],
+                        )
+                        if connection["interface_name"] != virtual_interface_name:
```

Comment:
I tighten up the check in https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/merge_requests/1223, so an interface-name is only allowed, if also a parent is allowed (mac-address alone is not enough).

In other words, if you didn't set parent (for a virtual infiniband profile), then the interface-name is also rejected.

Maybe this kind of "normalization" and checking is just pointless. The user is in fact not free to set an arbitary interface-name, it would seem easier to just reject any.

Reply:
As you mentioned, the normalization and checking in the NM are useless,  maybe the NM could have rejected the interface name set by user initially for the ipoib connection. Yes, currently, the simpliest approach in the role would be always rejecting the interface name as invalid for the ipoib connection.

Reply:
> Yes, currently, the simpliest approach in the role would be always rejecting the interface name as invalid for the ipoib connection.

The last version does not longer reject it (the validate() checks), or does it? Where does it happen? Despite the doc saying `**Note:** If the `p_key` is specified , then the `interface_name` must be unset.` and that libnm would really not like it.

Reply:
ah ok, `"the 'interface_name' must be unset for the ipoib "` . Never mind.